indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -239,27 +216,22 @@ public Response specGetAllHistory(@Context final HttpServletRequest req)
           @Override
           public Response apply(final SupervisorManager manager)
           {
-            final Map<String, List<VersionedSupervisorSpec>> supervisorHistory;
-            supervisorHistory = Maps.filterKeys(
-                manager.getSupervisorHistory(),
+            final Map<String, List<VersionedSupervisorSpec>> supervisorHistory = manager.getSupervisorHistory();
+
+            final Set<String> authorizedSupervisorIds = filterAuthorizedSupervisorIds(
+                req,
+                manager,
+                supervisorHistory.keySet()
+            );
+
+            final Map<String, List<VersionedSupervisorSpec>> authorizedSupervisorHistory = Maps.filterKeys(
```

Comment:
This object is unused. Something is wrong with this code.

Reply:
Thanks for noticing. https://github.com/druid-io/druid/pull/5459 fixes it.

Reply:
thanks for the catching that, opened a fix at #5460